packages/client/src/Dashboard.tsx

```diff
@@ -280,100 +281,114 @@ type Conversation = NonNullable<
-            const rowId = "conversation-row-" + id
-            return (
-              <React.Fragment key={index}>
-                <ListItem
-                  key={id}
-                  className={clsx(isRead && classes.read, classes.message)}
-                  alignItems="flex-start"
-                  onClick={() =>
-                    navigate(`/accounts/${accountId}/conversations/${id}`)
-                  }
-                  selected={isSelected}
-                >
-                  <ListItemAvatar>
-                    {isSelected ? (
-                      <MuiAvatar
-                        role="checkbox"
-                        aria-checked="true"
-                        aria-labelledby={rowId}
-                        className={classes.selectedAvatar}
-                        onClick={event => {
-                          event.stopPropagation()
-                          dispatch(Sel.unselect(id))
-                        }}
-                      >
-                        <CheckIcon />
-                      </MuiAvatar>
-                    ) : (
-                      <Avatar
-                        role="checkbox"
-                        aria-checked="false"
-                        aria-labelledby={rowId}
-                        onClick={event => {
-                          event.stopPropagation()
-                          dispatch(Sel.select(id))
-                        }}
-                        address={from}
-                      />
-                    )}
-                  </ListItemAvatar>
-                  <ListItemText
-                    id={rowId}
-                    primary={subject || "[no subject]"}
-                    secondary={
-                      <>
-                        <Typography
-                          component="span"
-                          variant="body2"
-                          className={classes.inline}
-                          color="textPrimary"
-                        >
-                          {moment(date).calendar()}
-                          {"\u00A0"}
-                          from{" "}
-                          {from
-                            ? from.name || `${from.mailbox}@${from.host}`
-                            : "unknown sender"}
-                        </Typography>
-                        {` — ${snippet}`}
-                      </>
-                    }
-                    classes={{
-                      secondary: classes.snippetText
-                    }}
-                  />
-                </ListItem>
-                {index === conversations.length - 1 ? (
-                  ""
-                ) : (
-                  <Divider variant="inset" component="li" />
-                )}
-              </React.Fragment>
-            )
-          }
-        )}
+        {conversations.map((conversation, index) => {
+          return (
+            <React.Fragment key={conversation.id}>
```

Comment:
I changed the `key` prop from `index` to `conversation.id`. React uses the `key` prop to match up elements that should be considered "the same" from one render to the next. When performing updates we want React to treat rows representing the same conversation as the same, and using the conversation ID as a key makes this work.